py/reporter_service.py

```diff
@@ -96,55 +96,88 @@ def parse_trace(self, post):
         return json.loads(params['json'][0])
     raise Exception('No json provided')
 
-  #parse the request because we dont get this for free!
-  def handle_request(self, post):
-    #get the reporter data
-    trace = self.parse_trace(post)
-
-    #lets get the uuid from json the request
-    uuid = trace.get('uuid')
-    if uuid is not None:
-      #do we already know something about this vehicleId already? Let's check Redis
-      partial = thread_local.cache.get(uuid)
-      if partial:
-        partial = pickle.loads(partial)
-        time_diff = trace['trace'][0]['time'] - partial[-1]['time']
-        #check to make sure time is not stale and not in future
-        if time_diff < os.environ.get('STALE_TIME', 60) and time_diff >= 0:
-          #Now prepend the last bit of shape from the partial_end segment that's already in Redis
-          #to the rest of the partial_start segment once it is returned from the segment_matcher
-          trace['trace'] = partial + trace['trace']
-    else:
-      return 400, 'No uuid in segment_match request!'
+  #report some segments to the datastore
+  def report(self, trace, partial):
+    uuid = trace['uuid']
+    trace_size = len(trace['trace'])
+    time_diff = trace['trace'][-1]['time'] - trace['trace'][0]['time']
+    #we buffer this point for now until we get more data
+    if not partial and (trace_size < os.environ.get('MAX_TRACE_SIZE', 120) or time_diff < os.environ.get('MAX_TIME_DIFF', 120)):
+      thread_local.cache.set(uuid, pickle.dumps(trace['trace']), ex=os.environ.get('PARTIAL_EXPIRY', 300))
+      return None
 
     #ask valhalla to give back OSMLR segments along this trace
     result = thread_local.segment_matcher.Match(json.dumps(trace, separators=(',', ':')))
     segments = json.loads(result)
 
-    #if there are segments
+    #TODO: there are a couple of problems below:
+    #if a probe just sits in the same place and continues getting a partial we'll do matching every time
```

Comment:
i feel confident about the distance thing so im going to do that. i think its intuitively a better measure than number of points